### PR TITLE
Add `virtio-rng` support

### PR DESCRIPTION
**Pull request by bchalios**
## Changes

This PR adds support for `virtio-rng` entropy device. 

## Reason

This would allow resolving issues with low entropy in the early boot stages since the kernel is using the device (when present) to gather additional entropy.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: src/devices/src/virtio/iovec.rs [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+            // region in the GuestMemoryMmap.
+            let iov_base = mem
+                .get_slice(desc.addr, desc.len as usize)?
+                .as_ptr()
+                .cast::<c_void>();
+            vecs.push(iovec {
+                iov_base,
+                iov_len: desc.len as size_t,
+            });
+            len += desc.len as usize;
```
Comment (roypat) [inline src/devices/src/virtio/iovec.rs:249]: Mhh, seems #3666 got lost in the rebase :(

Comment (bchalios) [reply src/devices/src/virtio/iovec.rs:249]: Damn! Let me find the commit and push it.

Comment (bchalios) [reply src/devices/src/virtio/iovec.rs]: Done!

**Review by roypat — DISMISSED** [review]
I had already reviewed each individual change of PR as they got introduced to the feature branch, excited to see this land! :)

**Pull request closed without merging**